src/cli.rs

```diff
@@ -1,13 +1,56 @@
-use std::{fs::File, io::Read, path::Path};
+use std::{
+    collections::HashSet,
+    num::{NonZeroUsize, ParseIntError},
+    path::Path,
+};
 
 use chrono::{DateTime, Local};
-use clap::{App, AppSettings, Arg};
-use flexi_logger::LevelFilter as LogLevel;
+use clap::{App, AppSettings, Arg, ArgMatches};
+use flexi_logger::{FlexiLoggerError, LevelFilter as LogLevel};
+use std::fs;
+use thiserror::Error;
 
-use crate::logs::init_logger;
+const DEFAULT_MAX_CONN: usize = 8;
 
-pub fn cli_init() -> AppConfig {
-    let app = App::new("paperoni")
+#[derive(derive_builder::Builder)]
+pub struct AppConfig {
+    /// Urls for store in epub
+    pub urls: Vec<String>,
+    pub max_conn: usize,
+    /// Path to file of multiple articles into a single epub
+    pub merged: Option<String>,
+    pub output_directory: Option<String>,
+    pub log_level: LogLevel,
+    pub can_disable_progress_bar: bool,
+    pub start_time: DateTime<Local>,
+    pub is_logging_to_file: bool,
+}
+
+#[derive(Debug, Error)]
+pub enum Error {
+    #[error("Failed to open file with urls: {0}")]
+    UrlFileError(#[from] std::io::Error),
+    #[error("Failed to parse max connection value: {0}")]
+    InvalidMaxConnectionCount(#[from] ParseIntError),
+    #[error("No urls for parse")]
+    NoUrls,
+    #[error("No urls for parse")]
+    AppBuildError(#[from] AppConfigBuilderError),
```

Comment:
This variant returns the same error message as `NoUrls` above. Should this be the case? I haven't used the `derive_builder` crate before so I don't know where the `AppConfigBuilderError` occurs in the code.

Reply:
Since in this case there are no default values for the fields, an error will occur if someone does not configure `AppConfig`